students/eric_rosko/lesson-04/mailroom.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python3
+
+'''
+Eric Rosko
+Session03
+mailroom.py
+Tuesday, Nov. 16, 2015
+Description:
+    Part Session 3 homework
+Requirements:
+    You must have py.test installed from http://pytest.org
+    python3 -m pip install pytest
+Usage:
+    py.test -v mailroom.py
+    or
+    python3 mailroom.py
+'''
+
+import json_save_meta as js
+from operator import *
+import io
+
+class DictionarySaver(js.JsonSaveable):
+    dict = js.Dict()
+    def __init__(self, dict):
+        self.dict = dict
+
+def send_thank_you(data):
+    isNameEntered = False
+    while not isNameEntered:
+        fullname = input("Enter full name: ")
+        if fullname == "list":
+            print(get_donor_names(data))
+        else:
+            isNameEntered=True
+
+    if not is_person_in_database(data,fullname):
+        print("User not in database - adding.")
+        data[fullname]=[]
+
+    isNumber = False
+    while not isNumber:
+        amount = input("Enter donation amount: ")
+        if verify_number(float(amount)):
+            isNumber=True
```

Comment:
I'd probably rewrite this as:
```
while True:
    try:
        amount = float(input("Enter donation amount: "))
    except TypeError:
        continue
```
Then you wouldn't need another function to verify the number.